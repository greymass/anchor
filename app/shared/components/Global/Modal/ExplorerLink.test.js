import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import GlobalModalExplorerLink from './ExplorerLink';

const mockProps = {
  t: text => text,
  blockExplorers: {
    'bloks.io': {
      txid: 'https://bloks.io/tx/{txid}',
      account: 'https://bloks.io/accounts/{account}',
    }
  },
  content: 'Click Here',
  linkData: 'teamgreymass',
  linkType: 'account',
  settings: { blockExplorer: 'bloks.io' }
};

describe('GlobalModalExplorerLink', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalModalExplorerLink {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
