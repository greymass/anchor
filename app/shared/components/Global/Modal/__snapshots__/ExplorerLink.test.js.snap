// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalModalExplorerLink renders correctly 1`] = `
<Translate(GlobalModalDangerLink)
  content="Click Here"
  link="https://bloks.io/accounts/teamgreymass"
  settings={
    Object {
      "blockExplorer": "bloks.io",
    }
  }
/>
`;
